google/cloud/storage/internal/async/connection_impl.cc

```diff
@@ -66,6 +67,25 @@ idempotency_policy(Options const& options) {
   return options.get<storage_experimental::IdempotencyPolicyOption>()();
 }
 
+std::unique_ptr<storage::internal::HashFunction> CreateHashFunction(
+    Options const& options) {
+  auto const crc32c =
+      !options.has<storage_experimental::UseCrc32cValueOption>() ||
+      options.get<storage_experimental::EnableCrc32cValidationOption>();
+  auto const md5 =
+      !options.has<storage_experimental::UseMD5ValueOption>() ||
+      options.get<storage_experimental::EnableMD5ValidationOption>();
+
+  if (!crc32c && !md5) return storage::internal::CreateNullHashFunction();
+  if (crc32c && md5) {
+    return std::make_unique<storage::internal::CompositeFunction>(
+        std::make_unique<storage::internal::Crc32cHashFunction>(),
+        storage::internal::MD5HashFunction::Create());
+  }
+  if (crc32c) std::make_unique<storage::internal::Crc32cHashFunction>();
+  return storage::internal::MD5HashFunction::Create();
```

Comment:
Opt nit:  I'd make `return storage::internal::CreateNullHashFunction()` the default return by guarding this with `if (md5)` and deleting line 79.

Reply:
Done. Also note the bug in line 85, I will be adding tests in an upcoming PR.

Reply:
Do let me know if I should reorder the PRs to include this code and the tests in a single PR.

Reply:
I convinced myself that I should also include the tests.  Going to pause this PR until those are ready to merge.

Reply:
Okie dokie.